test/test-Queue.js

```diff
@@ -0,0 +1,42 @@
+"use strict";
+
+const {Queue} = require("lib/task-queue/Queue");
+
+exports["test enqueuing and dequeuing"] = function(assert, done) {
+  let queue = new Queue();
+
+  assert.equal(queue.size, 0, "Queue start size is 0.");
+
+  let items = new Set(Array(100).keys());
+
+  for (let item of items) {
+    queue.enqueue(item);
+  }
+
+  assert.equal(queue.size, items.size, "queue size is correct.");
+
+  let item = queue.dequeue();
+  while (item !== null) {
+    assert.ok(items.delete(item), "Dequeued item from the original set.");
+    item = queue.dequeue();
```

Comment:
We should test a property of this type of `Queue` as well, that it behaves in LIFO fashion.